### PR TITLE
Prevent Mount for deployment volumes to be reset on name change

### LINKED ISSUE #5317: UI allows Deployments > Storage yaml creation and mountPath name
- Mount Point (mountPath in yaml) is a required field, but doesn't show they key in yaml until the field is populated.
- Editing the Volume Name erases Mount Point input.

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #5317
While creating a deployment storage volume in a cluster, the Mount Point value is reset on Volume name change.

### Occurred changes and/or fixed issues
- Corrected issue which made the whole Mount list load again with no values
- Added required `*` symbol for every Volume input field

### Technical notes summary
- Improved readability with comments, renaming and formatting code
- Allowed Select component to be flagged as required by adding prop, UI functionality already existing
- Flagged all the required fields for every Volume type
- Prevent Codemirror from float over whole page (component has been used to inspect code changes, separated issue already exists)
- Added unit tests for Mount component
- Removed deprecated from logic prop
- Removed Ephemeral component, after agreement with @mantis-toboggan-md 
- Removed key from Mount component, generating a new component due DOM identifier value not matching
- Added `remove()` emission for ArrayListGroup and ArrayList 
- Added cleanup method to remove mapped Mount for given names when Volume are removed (were causing bleeding in other Volumes)

### Areas or cases that should be tested
As defined in issue, plus:
- Add a volume
- Define a mount point
- Repeat it 3 times (due cases different cases)
- Delete previous volumes
 
### Areas which could experience regressions
VolumeClaimTemplate component.

#### Editing names

https://user-images.githubusercontent.com/5009481/162948095-d8c99148-1f2c-4fb0-a3d6-b65363a179d0.mov

#### Adding/removing volumes

https://user-images.githubusercontent.com/5009481/164090163-03560101-9d26-4537-a58a-f0689c787502.mov

### CODE REVIEW

**Comment by cnotv** [comment]
Test for this very issue could not be completed due an issue with Jest #5631

**Review by mantis-toboggan-md — APPROVED** [review]
LGTM - The 'Add Volume' dropdown caret looks weird, but I see you've already filed an issue for that

File: edit/workload/storage/index.vue [hunk 1]
```diff
@@ -281,13 +299,22 @@ export default {
             />
           </div>
         </div>
-        <Mount :key="props.row.value.name" :container="container" :pod-spec="value" :name="props.row.value.name" :mode="mode" />
+
+        <!-- Mount point list to be mapped to volume -->
+        <Mount
+          :container="container"
```
Comment (mantis-toboggan-md) [inline edit/workload/storage/index.vue:305]: Removing the Mount key causes issues when deleting a volume. Try:

1. Add a volume
2. Define a mount point
3. Add a second volume
4. Define a mount point for that volume
5. Delete the first volume

Observe that the second (now only) volume lists both mount points

Comment (richard-cox) [reply edit/workload/storage/index.vue:305]: @mantis-toboggan-md Should this issue be resolved in this PR or should it be tackled via a separate (github) issue?

Comment (mantis-toboggan-md) [reply edit/workload/storage/index.vue:305]: It'd certainly be preferable not to knowingly introduce a bug while fixing another, but I've just done a little investigation myself and it looks like it wont be a quick fix to resolve both #5317 and what I've described. Given that the former is more obnoxious and our deadline approaches, I think it's probably fine to track the latter separately and fix it in 2.6.6.

Comment (cnotv) [reply edit/workload/storage/index.vue:305]: I can try to generate an identifier which is not paired with the name.

Comment (cnotv) [reply edit/workload/storage/index.vue:305]: It seems like before it was always removing every mount, while now is not, but now I got some race conditions while creating partial copies or references of the original value/container/volume.
Working on it.

**Review by mantis-toboggan-md — APPROVED** [review]
LGTM

**Pull request merged**